### PR TITLE
[#547] Define "results" for the "Up" navigation to deliver result to previous screens

### PR DESCRIPTION
**Pull request by luongvo**
Closes https://github.com/nimblehq/android-templates/issues/547

## What happened 👀

- Define "results" for the "Up" navigation to deliver results to previous screens.
- Add a sample to show the `Updated!` message on Home screen after clicking on the `Update` button on the Second screen.

## Insight 📝

- To navigate back to the previous screen with a result, call this for example:

	```
	navigator(BaseDestination.Up().addResult(KeyResultOk, true))
	```

## Proof Of Work 📹

https://github.com/nimblehq/android-templates/assets/16315358/8953291a-88f3-4083-ba9b-d5659666ccc5

### CODE REVIEW

**Review by kaungkhantsoe — APPROVED** [review]
LGTM

File: sample-compose/app/src/main/java/co/nimblehq/sample/compose/extensions/SavedStateHandleExt.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package co.nimblehq.sample.compose.extensions
+
+import androidx.lifecycle.SavedStateHandle
+
+fun <T> SavedStateHandle.getAndRemove(key: String): T? {
```
Comment (lydiasama) [inline sample-compose/app/src/main/java/co/nimblehq/sample/compose/extensions/SavedStateHandleExt.kt:5]:
```suggestion
fun <T> SavedStateHandle.getThenRemove(key: String): T? {
```
I think we better use `then` word.

File: sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/AppDestination.kt [hunk 2]
```diff
@@ -16,7 +17,7 @@ sealed class AppDestination(val route: String = "") {
 
     open var parcelableArgument: Pair<String, Any?> = "" to null
 
-    object Up : AppDestination()
+    data class Up(val results: List<Result> = emptyList()) : AppDestination()
```
Comment (lydiasama) [inline sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/AppDestination.kt:20]:
```suggestion
    data class Up(val results: Map<String, Any> = emptyList()) : AppDestination()
```
Why don't we use `Map` instead? 🤔

Comment (luongvo) [reply sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/AppDestination.kt]: @lydiasama good suggestion, fixed in https://github.com/nimblehq/android-templates/pull/530/commits/54ef3e3b4700b5aaa04f10fb134876ce54c3ac48

File: sample-compose/app/src/main/java/co/nimblehq/sample/compose/extensions/SavedStateHandleExt.kt (hunk 1 above)
Comment (luongvo) [reply sample-compose/app/src/main/java/co/nimblehq/sample/compose/extensions/SavedStateHandleExt.kt]: fixed in https://github.com/nimblehq/android-templates/pull/530/commits/4ed2969da6824ce98cefed38d01dedb7c7a5d788

**Review by ryan-conway — APPROVED** [review]

**Review by lydiasama — APPROVED** [review]

**Review by chornerman — APPROVED** [review]

**Pull request merged**